### PR TITLE
Bluetooth: tester: Add Mesh Network Send command

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This adds Mesh Network Send command implementation.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

### CODE REVIEW

File: tests/bluetooth/tester/src/mesh.c [hunk 1]
```diff
@@ -479,6 +485,51 @@ static void lpn_poll(u8_t *data, u16_t len)
...
+	};
+	struct bt_mesh_net_tx tx = {
+		.ctx = &ctx,
+		.xmit = bt_mesh_net_transmit_get(),
+		.sub = bt_mesh_subnet_get(net.net_idx),
+	};
+	u8_t status = BTP_STATUS_SUCCESS;
+	int err;
+
+	ctx.send_ttl = cmd->ttl;
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/mesh.c]: It's a bit disconcerting that the handler is given a 'len' parameter, but you never use it and instead fully trust that you can safely go and dereference cmd here like this. Is the minimum length and that "len == sizeof(*cmd) + cmd->payload_len" checked somewhere else?

File: tests/bluetooth/tester/src/mesh.c [hunk 2]
```diff
@@ -13,6 +13,11 @@
 #include <misc/byteorder.h>
 #include "bttester.h"
 
+#include "../../../../subsys/bluetooth/host/mesh/foundation.h"
+#include "../../../../subsys/bluetooth/host/mesh/mesh.h"
+#include "../../../../subsys/bluetooth/host/mesh/net.h"
+#include "../../../../subsys/bluetooth/host/mesh/transport.h"
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/mesh.c]: Did you consider having the tester register a dummy client model and do the sending of messages using public APIs as if the messages belong to that model? That'd let us avoid having to include private headers.

**Review by jhedberg — CHANGES_REQUESTED** [review]

File: tests/bluetooth/tester/src/mesh.c (hunk 1 above)
Comment (MariuszSkamra) [reply tests/bluetooth/tester/src/mesh.c]: no, it is never checked...

File: tests/bluetooth/tester/src/mesh.c (hunk 2 above)
Comment (MariuszSkamra) [reply tests/bluetooth/tester/src/mesh.c]: you mean to send via bt_mesh_model_send? I will try.

Comment (jhedberg) [reply tests/bluetooth/tester/src/mesh.c]: yes, that's what I meant

Comment (jhedberg) [reply tests/bluetooth/tester/src/mesh.c]: note that you'll probably want to use a vendor model, e.g. with CID 0xffff

Comment (jhedberg) [reply tests/bluetooth/tester/src/mesh.c]: and I just realized you even have a define for that already: CID_LOCAL

Comment (MariuszSkamra) [reply tests/bluetooth/tester/src/mesh.c]: Yes, I added this for composition data.

Comment (MariuszSkamra) [reply tests/bluetooth/tester/src/mesh.c]: @jhedberg The problem is I don't have app_key. Should I add application key, just like you did in mesh_demo? Using BT_MESH_KEY_DEV failed for me with error:
"model_send: Model not bound to AppKey 0xfffe".
I tried also BT_MESH_KEY_UNUSED, but then bt_mesh_app_key_find() returns NULL, and send fails with -EINVAL error.

**Comment by MariuszSkamra** [comment]
@jhedberg I managed to do this via bt_model_send

**Review by jhedberg — APPROVED** [review]
I definitely like this approach better since it avoids using internal APIs. I'll wait for a second review from @sjanc before merging.

**Review by sjanc — APPROVED** [review]
looks good

**Review by jhedberg — APPROVED** [review]

**Comment by jhedberg** [comment]
Could you sort out the "This branch has conflicts that must be resolved" issue?

**Comment by MariuszSkamra** [comment]
rebased

**Review by sjanc — APPROVED** [review]

**Pull request merged**